packages/core/src/State.ts

```diff
@@ -110,21 +110,39 @@ interface MachineSnapshotBase<
 
   /**
    * Whether the current state value is a subset of the given parent state value.
-   * @param parentStateValue
+   * @param testValue
    */
   matches: <
     TSV extends TResolvedTypesMeta extends TypegenEnabled
       ? Prop<Prop<TResolvedTypesMeta, 'resolved'>, 'matchesStates'>
       : StateValue
   >(
-    parentStateValue: TSV
+    this: MachineSnapshotBase<
```

Comment:
I tried to just use `this: this` but unfortunately TS doesn't like it: [TS playground](https://www.typescriptlang.org/play?ts=5.4.0-dev.20231110#code/JYOwLgpgTgZghgYwgAgEJwM4oN4ChnIAmA9gMpgCuMMAXMgBRgAWwGdzrAlMgLwB8yAG7FghANy4AvrlyhIsRCgCCCMMEEoIAD0ghCGNJhz5kGMHEptkAIkRqN1idNnho8JMgAixEJp0Q9A3QsZDwCMwsKK2sSX0cpGTAATwAHFHILFB5kFXsUAB8vHwgJXEIIBAAbOCgUBB8zZBhiYjoMyFLm4gA6EnIqGHpOCSA)

It's also not mandatory at all to have `this` here in the annotation but it nicely guards people against mistakes like this: [TS playground](https://www.typescriptlang.org/play?ts=5.4.0-dev.20231110#code/JYOwLgpgTgZghgYwgAgMpjpA6sMALAFT2AGdkBvAKGWQFtME8IT1MIAKfUgLjQ21yFiJADTIAbnAA2AVwi8SYKKADmASl4AjAPbapEOCEoBfSpQAmEBFLhQUCbSEXJFbAIy9WA-EVKUHTmB0DEwkbsgAvC78EG4AdPRgjMxeEJSJyWHsAOTZamagkLCIKKk4+NoyYL5kVDQZoanskrLy0cog6lq6+oYmZpbWtvaOzq6QAEyeMeV4ldXC-qNBDcwTkdFsEwkhKTHpuyQTOXlAA)